Makefile

```diff
@@ -271,38 +279,47 @@ $(foreach prog,$(PROGRAMS),$(eval $(call BUILD_ASSEMBLY,$(prog))))
 
 ########################## MAKE/INSTALL RULES ##########################
 #
-default: dependencies core
+default: core
 
 core: game renderers mods utility
 
 tools: editor gamemonitor
 
-package: cli-dependencies core editor gamemonitor docs version
+package: all-dependencies core tools docs version
 
 mods: mod_common mod_ra mod_cnc mod_d2k mod_ts
 
-all: cli-dependencies core tools
+all: dependencies core tools
 
 clean:
-	@-$(RM_F) *.exe *.dll ./OpenRA*/*.dll ./OpenRA*/*.mdb *.mdb mods/**/*.dll mods/**/*.mdb *.resources
+	@-$(RM_F) *.exe *.dll *.dylib *.config ./OpenRA*/*.dll ./OpenRA*/*.mdb *.mdb mods/**/*.dll mods/**/*.mdb *.resources
 	@-$(RM_RF) ./*/bin ./*/obj
-	@-$(RM_F) ./thirdparty/*.dll
-	@-$(RM_RF) ./thirdparty/windows
+	@-$(RM_RF) ./thirdparty/download
 
 distclean: clean
 
-dependencies: cli-dependencies native-dependencies
+cli-dependencies:
+	@./thirdparty/fetch-thirdparty-deps.sh
+	@ $(CP_R) thirdparty/download/*.dll .
+	@ $(CP_R) thirdparty/download/*.dll.config .
+	@ $(CP) thirdparty/SDL2-CS.dll.config .
+
+linux-dependencies: cli-dependencies linux-native-dependencies
+
+linux-native-dependencies:
+	@./thirdparty/configure-native-deps.sh
 
 windows-dependencies:
-	cd thirdparty && ./fetch-thirdparty-deps-windows.sh && cd ..
+	@./thirdparty/fetch-thirdparty-deps-windows.sh
 
-cli-dependencies:
-	cd thirdparty && ./fetch-thirdparty-deps.sh && cd ..
-	@ $(CP_R) thirdparty/*.dll .
-	@ $(CP_R) thirdparty/*.dll.config .
+osx-dependencies: cli-dependencies
+	@./thirdparty/fetch-thirdparty-deps-osx.sh
+	@ $(CP_R) thirdparty/download/osx/*.dylib .
+	@ $(CP_R) thirdparty/download/osx/*.dll.config .
 
-native-dependencies:
-	@./thirdparty/configure-native-deps.sh
+dependencies: $(os-dependencies)
+
+all-dependencies: cli-dependencies windows-dependencies osx-dependencies
```

Comment:
We want to make sure we don't end up packaging linux/windows deps in the OSX app bundle, and so on.